docs/sources/flow/reference/components/module.file.md

```diff
@@ -42,20 +43,35 @@ Name | Type | Description | Default | Required
 `detector`       | `string`   | Which file change detector to use (fsnotify, poll) | `"fsnotify"` | no
 `poll_frequency` | `duration` | How often to poll for file changes | `"1m"` | no
 `is_secret`      | `bool`     | Marks the file as containing a [secret][] | `false` | no
-`arguments`      | `map(any)` | The values for the supported arguments in the module contents. | | no
-
-`arguments` allows us to pass parameterized input into a module. The values
-passed in `arguments` correspond to [argument blocks][] defined in the module
-source.
-
-An `argument` marked non-optional in the module being loaded is required in the
-`arguments`. It is also not valid to provide an `argument` not defined in the
-module being loaded.
 
 [secret]: {{< relref "../../config-language/expressions/types_and_values.md#secrets" >}}
 
 {{< docs/shared lookup="flow/reference/components/local-file-arguments-text.md" source="agent" >}}
 
+## Blocks
+
+The following blocks are supported inside the definition of `module.file`:
+
+Hierarchy        | Block      | Description | Required
+---------------- | ---------- | ----------- | --------
+arguments | [arguments][] | Arguments to pass to the module. | no
+
+[arguments]: #arguments-block
+
+### arguments block
+
+The `arguments` block specifies the list of arguments to pass to the loaded
```

Comment:
Thoughts on going with:

"The `arguments` block specifies the list of values to pass to the loaded"

I hate using the same word in the definition of the word ;)

Reply:
Works for me 👍